torch/distributed/_composable/fsdp/_fsdp_init.py

```diff
@@ -28,3 +34,27 @@ def _init_default_fully_shard_mesh(device_type: str) -> DeviceMesh:
     default_pg = dist.distributed_c10d._get_default_group()
     mesh = init_device_mesh(device_type=device_type, mesh_shape=(default_pg.size(),))
     return mesh
+
+
+def _get_managed_modules(root_module: nn.Module) -> List[nn.Module]:
+    modules: List[nn.Module] = []
+    # Track visisted modules to avoid visiting shared modules multiple times
+    visited_modules: Set[nn.Module] = set()
+
+    def dfs(module: nn.Module) -> None:
```

Comment:
I feel we can just loop through the `root_module.named_modules()` API where it already gives a dfs order? we don't need to keep the visited module logic if we just reuse that

Reply:
If we use `named_modules()`, since that is a generator, I am not sure how to skip the child modules of a `replicate` module. (E.g., there could be modules after the `replicate` subtree after that we should not skip.)